tests/impl.cpp

```diff
@@ -1442,7 +1442,7 @@ result_t test_mm_cvtps_pi16(const SSE2NEONTestImpl &impl, uint32_t iter)
     int16_t rnd[4];
 
     for (int i = 0; i < 4; i++) {
-        if (INT16_MAX <= _a[i] && _a[i] <= INT32_MAX) {
+        if (INT16_MAX <= (int) _a[i] && (int) _a[i] <= INT32_MAX) {
```

Comment:
I prefer to cast the `INT16_MAX` to `float` since casting with `int` would have data loss.